pkg/tnf/pkg/jobs/jobcontroller_test.go

```diff
@@ -27,13 +36,305 @@ const (
 	controllerName   = "TestJobController"
 	operandName      = "dummy-controller"
 	operandNamespace = "openshift-test"
+	// From github.com/openshift/library-go/pkg/operator/resource/resourceapply/batch.go
+	specHashAnnotation = "operator.openshift.io/spec-hash"
 )
 
-// fakeOperatorInstance is a fake Operator instance that  fullfils the OperatorClient interface.
-type fakeOperatorInstance struct {
-	metav1.ObjectMeta
-	Spec   opv1.OperatorSpec
-	Status opv1.OperatorStatus
+var (
+	conditionAvailable   = controllerName + opv1.OperatorStatusTypeAvailable
+	conditionProgressing = controllerName + opv1.OperatorStatusTypeProgressing
+)
+
+func TestJobCreation(t *testing.T) {
```

Comment:
There are quite a few similarities between this TestJobCreation and some logic in TestSync, is it necessary?